exchanges/bitfinex/bitfinex_test.go

```diff
@@ -1180,53 +1212,49 @@ func TestWsSubscribe(t *testing.T) {
 	assert.True(t, ok, "sub.Key should be an int")
 
 	err = b.Unsubscribe(subs)
-	assert.ErrorIs(t, err, stream.ErrUnsubscribeFailure, "Unsubscribe should error")
 	assert.ErrorContains(t, err, strconv.Itoa(chanID), "Unsubscribe should contain correct chanId")
 	assert.ErrorContains(t, err, "unsubscribe: invalid (code: 10400)", "Unsubscribe should contain correct upstream error")
 
 	err = b.Subscribe(subscription.List{{
-		Channel: wsTicker,
+		Channel: subscription.TickerChannel,
 		Pairs:   currency.Pairs{currency.NewPair(currency.BTC, currency.USD)},
 		Asset:   asset.Spot,
-		Params:  map[string]interface{}{"key": "tBTCUSD"},
+		Params:  map[string]any{"key": "tBTCUSD"},
 	}})
-	assert.ErrorIs(t, err, stream.ErrSubscriptionFailure, "Trying to use a 'key' param should error ErrSubscriptionFailure")
 	assert.ErrorIs(t, err, errParamNotAllowed, "Trying to use a 'key' param should error errParamNotAllowed")
 }
 
-// TestSubscribeReq tests the channel to request map marshalling
-func TestSubscribeReq(t *testing.T) {
-	c := &subscription.Subscription{
-		Channel: wsCandles,
-		Pairs:   currency.Pairs{currency.NewPair(currency.BTC, currency.USD)},
-		Asset:   asset.MarginFunding,
-		Params: map[string]interface{}{
-			CandlesPeriodKey: "30",
-		},
+// TestSubToMap tests the channel to request map marshalling
+func TestSubToMap(t *testing.T) {
+	s := &subscription.Subscription{
+		Channel:  subscription.CandlesChannel,
+		Asset:    asset.Spot,
+		Pairs:    currency.Pairs{currency.NewPair(currency.BTC, currency.USD)},
+		Interval: kline.OneMin,
 	}
-	r, err := subscribeReq(c)
-	assert.NoError(t, err, "subscribeReq should not error")
-	assert.Equal(t, "trade:1m:fBTCUSD:p30", r["key"], "key contain period and default timeframe")
 
-	c.Params = map[string]interface{}{
-		CandlesTimeframeKey: "15m",
-	}
-	r, err = subscribeReq(c)
-	assert.NoError(t, err, "subscribeReq should not error")
-	assert.Equal(t, "trade:15m:fBTCUSD", r["key"], "key should be contain specific timeframe and no period")
+	r := subToMap(s, s.Asset, s.Pairs[0])
+	assert.Equal(t, "trade:1m:tBTCUSD", r["key"], "key should be contain specific timeframe and no period")
+
+	s.Interval = kline.FifteenMin
+	s.Asset = asset.MarginFunding
+	s.Params = map[string]any{CandlesPeriodKey: "p30"}
+
+	r = subToMap(s, s.Asset, s.Pairs[0])
+	assert.Equal(t, "trade:15m:fBTCUSD:p30", r["key"], "key contain period and no timeframe")
```

Comment:
```suggestion
	assert.Equal(t, "trade:15m:fBTCUSD:p30", r["key"], "key should contain a period and specific timeframe")
```
Since the interval of 15m is included

Reply:
Fixed 3148e6f3